Assets/Scripts/Enemy/EnemyService.cs

```diff
@@ -43,8 +46,12 @@ public void SpawnEnemyTank()
             enemySpawnPos = enemySpawnPos.SetRandomVectorXYZ(-39, 39, 0, 10, -39, 39);
             enemySpawnPos = enemySpawnPos.SetY(-3.68f);
             EnemyTankPositions.Add(enemySpawnPos);
-            GameObject enemyTank = Instantiate(EnemyTankViews[Random.Range(0, EnemyTankViews.Count)], enemySpawnPos, Quaternion.identity) as GameObject;
-            InstantiatedEnemyTanks.Add(enemyTank);
+            GameObject enemyTank = enemyPoolService.GetEnemyTank(EnemyTankViews[Random.Range(0, EnemyTankViews.Count)]);
+            enemyTank.transform.position = enemySpawnPos;
+            enemyTank.transform.rotation = Quaternion.identity;
+            enemyTank.SetActive(true);
```

Comment:
setting position/rotation/enable/disable should be functions on the controller and not done in the service level

Reply:
No Controller made for enemy behavior kept it simple so I used the review and corrected bullet and player logic as added  reset position() logic  to the controller